tools/system_libs.py

```diff
@@ -1380,6 +1380,9 @@ def get_files(self):
         path='system/lib/wasm_worker',
         filenames=['library_wasm_worker.c' if self.is_ww or self.is_mt else 'library_wasm_worker_stub.c'])
 
+  def can_use(self):
+    return super().can_use() and self.is_ww
```

Comment:
It looks like this library provides `_stub` variant which can be used when WASM_WORKERS is not set.   I'm not sure why this would be needed but it does seems like that should be the one that gets used.

Reply:
@sbc100 

```
def get_base_name(self):
    name = 'libwasm_workers'
    if not self.is_ww and not self.is_mt:
      name += '_stub'
    if self.debug:
      name += '-debug'
    return name
```
Indeed there is a `_stub` here but it is used only when not `ww` and `mt`.  
If `mt` is enable with wasm_workers linked https://github.com/emscripten-core/emscripten/issues/19482 will happen.

Or maybe we can disable  the ctor in `library_wasm_workers.c` when pthreads enabled?
```
__attribute__((constructor(48)))
static void emscripten_wasm_worker_main_thread_initialize() {
	uintptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
	assert((*sbrk_ptr & 15) == 0);
	assert(__builtin_wasm_tls_align() <= 16);
	__wasm_init_tls((void*)*sbrk_ptr);
	*sbrk_ptr += (__builtin_wasm_tls_size() + 15) & -16;
}
```